### DISCUSSION TITLE
Is there any approach to set max_running_time of job?

### DISCUSSION THREAD

**Discussion by chibai** [agronholm/apscheduler#584] [Q&A] [answered]
I'm trying to set a maximum time of a job, is there any approach?
For example, I'm want the job killed after running 5 minutes even the job is not finished.

**Comment by agronholm** [comment, marked as answer]
No, you cannot (safely) kill threads on Python. There is a method coming in v4 that lets synchronous jobs check if they have been externally cancelled but that requires modifying the job code to support that.
